### DISCUSSION TITLE
Project Update July 2023

### DISCUSSION THREAD

**Discussion by DeeEmm** [DIY-Flow-Bench/DIY-Flow-Bench#155] [Announcements]
There's no code changes so far this month. I've been a bit tied up with other work. We are also waiting for those contributors with prototype PCBs to finish builds and start submitting data. All prototype PCBs have now been sent out. The next stage is to collect and correlate data from the testers so that the code can be validated and finalised.

We also now have a contributor on board that's using an orifice style bench, so this will be a valuable source for implementing and testing the orifice bench code and provide a pathway for others wanting to convert orifice style benches.

**Comment by black-top** [comment]
Since we have to compare the measurement results, we must not forget the discussion about the atmospheric compensation standard. We need to decide which standard would be the most suitable for us to use. As we discussed earlier, probably one of the  standards used for dyno.

**Reply by DeeEmm** [reply]
Yes I will have to find the conversation thread but the dyno compensation standard seems the most logical to use.

Regarding data, as long as the baro, humidity and temp are also recorded we can make that conversion afterwards. I would prefer to receive raw data this way so that I can then record it in a spreadsheet and apply the same conversation factor to all results. This will hopefully eliminate any discrepancies created from the conversion process itself. 

If I get a chance I will take a look at including the conversion calcs in the code, which will achieve the same thing.

**Reply by black-top** [reply]
same problem with "top brand" flow bench https://www.youtube.com/watch?v=eh9TLRPa3NQ

**Reply by black-top** [reply]
I'm finally getting back to the project, so much time has passed that I don't even remember where I stopped. I don't remember if we talked about it, but the flow data capturing and display in the graph works properly, but when saving the graph image, a file is created that is empty. I don't know if there is a bug in the code, or maybe the problem is that I haven't connected the sd card module yet?
How are you? Is there any possibility to return to the project any time soon? Is there any feedback data from other people's measurements?

**Reply by DeeEmm** [reply]
Hi @black-top so far I have received no data from anyone, so the project is just sitting waiting for data to be submitted before I can progress. 

It's been a while, so working from memory here. IIRC the file should be saved to SPIFFs in JSON format. You can use the file browser to access it and download it. It should be named liftdata.json

**Reply by black-top** [reply]
Yes it saves liftdata.json and it show values correctly in text format when I open it with notepad, but image LiftGraph.png show 12kb size and 800x750 pix, but when the file is opened it is empty, there is no image, the opened image shows as a transparent rectangle. A simple solution for now is just a screenshot, but probably there is a bug in code, because same problem on phone and pc when image is saved. 
Another small detail that would be convenient is the possibility to change not only the lift interval but also the maximum lift and maximum flow values on the graph in the configuration. Now the graph looks out of proportion with the maximum 1000cfm when measuring for example up to 350cfm.

**Reply by DeeEmm** [reply]
I've raised it as an issue. I'll have to take a deeper look into it as I cannot recall from memory the status of this. It may simply be that it is yet to be finished. The data-logging stuff is actually roadmapped for the V3 code (we are still on V2). I pulled it forwards as it gave me something to do whilst waiting for beta test data. I _thought_ that it was working,  but maybe it isn't

https://github.com/DeeEmm/DIY-Flow-Bench/issues/157

**Reply by black-top** [reply]
Another small thing is that there is no option to select the capture mode for adjusted/unadjusted flow. Now it is only capturing the unadjusted flow. 
It would also be good if a table with lift and flow values was shown under the graph, because the exact values are not clearly visible on the graph

**Reply by DeeEmm** [reply]
This is intentional. Recorded data is essentially the raw data. The intent was that any manipulation of the data can be done in Excel.

However it would be easy enough to add an option so that the end user can change this.

I will add it as a feature request.

**Reply by black-top** [reply]
I don't want to bother you, but after coming back to the project, I have a lot of ideas.

Will be nice to have feature for pito input calibration for air speed probing so it will display air speed in m/s or ft/s instead of differential pressure if pito input is unused with maf or orifice style flow bench.

 My small pump starts to lose pressure after approximately 150-160cfm @ 28in H2o, so corrected and uncorrected flow numbers start to diverge,  so I probably will change to my big pump that was too powerful for my electrical installation, but now I have VFD, so it should be ok. 

I still need to fix few leaks and do calibrations for flow bench.

**Reply by DeeEmm** [reply]
Not bothering me at all. :) Always good to get different ideas. We can capture them and add them to the roadmap.

One question. How do you think that we can calibrate the air flow speed?

I had considered that the pitot probe just shows area of higher or lower pressure than the reference source as a tool to aid with determining . It is possible to equate this to flow but calibration is I think a bit of a challenge.

I will create an issue and add it to the roadmap.

**Reply by black-top** [reply]
Well, it looks like there are no easy way to calibrate it, but thinking in terms of head porting, exact velocity does not matter so much. The biggest advantage, I think, will be the ability to see port velocity in real time by probing the port for looking  high speed and dead spots of port with a pitot probe. So probably the option to select just m/s or ft/s in the configuration will be good enough.  

The conversion from pitot pressure to velocity looks very simple, and we already have density from the BME sensor  https://www.grc.nasa.gov/www/k-12/VirtualAero/BottleRocket/airplane/pitot.html
One thing  I see that may be a problem if a sensor on board is used for a pitot tube  with relatively long tubes is that it might affect readings, but if the sensor will be used close to the pitot probe and  relatively long wires can also disrupt signal, so maybe a different sensor with digital communication is the best option in this case. 

By the way, in the liftdata.json file, there are no barometric data captured from BME.

**Reply by DeeEmm** [reply]
Yes the lift data file only contains lift data at this point in time. Data logging was not actually scheduled until the V3 release but I added a basic version to V2 make the project more usable. It will in time be expanded to include additional data and features.

**Reply by black-top** [reply]
Ok, I just think that at this development and testing stage, it is important for comparing different flow bench data. Maybe just a single time BME data capture at the first lift point data capture, or just update the same atmospheric variables in the liftdata.json after each lift capture, so the last data capture will have the latest atmospheric data because it does not change much during a single test session.

**Reply by DeeEmm** [reply]
I agree. It would be useful to have this data to compare results. 

This actually flows on to the other conversation relating to recording data against a reference standard. If recorded data is transposed against standardised environmental values, all results would be directly comparable, no matter where they were taken.

The discussion was to use the same SAE / DIN standards used by automative manufacturers for dyno power correction when measuring / quoting power figures. This at least provided some continuity between air density at design time (the flow bench) and air density in real time usage (the dyno). 

Ultimately results would be transposed against standardised Baro / Humidity / Temperature to give a Standardised output. However, I think that it may be be prudent to record the raw data, including environmental measurements in the interim so that any such conversion calcs can be validated.

I will take a look into this but it will not be until some time into the new year as I'm pretty tied up with renovations at the moment.

**Reply by DeeEmm** [reply]
This was the conversation. Or at least the notes I made on it...

https://github.com/users/DeeEmm/projects/4?pane=issue&itemId=26542224

**Reply by black-top** [reply]
I completely forgot about it, now everything is clear why it is so

**Reply by DeeEmm** [reply]
Hehe. I had to refresh my memory. Been a while since I looked at this.
